editor/editor_properties.cpp

```diff
@@ -2183,7 +2183,55 @@ void EditorPropertyResource::_menu_option(int p_which) {
 		case OBJ_MENU_NEW_SCRIPT: {
 
 			if (Object::cast_to<Node>(get_edited_object())) {
-				EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				if (get_edited_property() == "script") {
+					EditorNode::get_singleton()->get_scene_tree_dock()->open_script_dialog(Object::cast_to<Node>(get_edited_object()));
+				} else {
+					// Copied from editor/scene_tree_dock.cpp#L380 and edited for our purposes
+					// There's a good chance we could merge this back into that code with some modifications
+					//Ref<Script> existing = get_edited_object()->get(get_edited_property());
+					//Script *existing = Object::cast_to<Script>(get_edited_object()->get(get_edited_property()));
+					RES existing = get_edited_object()->get(get_edited_property());
+
+					String inherits;
+					if (existing.is_valid()) {
+						//inherits = existing->get_class();
+						for (int i = 0; i < ScriptServer::get_language_count(); i++) {
+							ScriptLanguage *l = ScriptServer::get_language(i);
+							if (l->get_type() == existing->get_class()) {
+								String name = l->get_global_class_name(existing->get_path());
+								if (ScriptServer::is_global_class(name) && EDITOR_GET("interface/editors/derive_script_globals_by_name").operator bool()) {
+									inherits = name;
+								} else if (l->can_inherit_from_file()) {
+									inherits = "\"" + existing->get_path() + "\"";
+								}
+								break;
+							}
+						}
+					} else {
+						inherits = "Resource";
+					}
+
+					Node *edited_node = Object::cast_to<Node>(get_edited_object());
+
+					String path = edited_node->get_filename().get_basename();
+					if (path == "") {
+						String root_path = EditorNode::get_singleton()->get_editor_data().get_edited_scene_root()->get_filename();
+						if (root_path == "") {
+							path = String("res://").plus_file(edited_node->get_name());
+						} else {
+							path = root_path.get_base_dir().plus_file(edited_node->get_name());
+						}
+					}
+					// path = <node_or_scene_name>.<property_name>.
+					// The extra period at the end is to prevent the property name from being seen as the type by ScriptCreateDialog
+					path = path + "." + get_edited_property() + ".";
```

Comment:
The script path is treated the same as with regular scripts, with the exception of the property name being added to the end of the script path. So for example, if I'm attaching a script to a property named "idle_state" on a node named "player", the script path will default to `player.idle_state.` (with an ending period to prevent the property name from being seen as an extension).